### PR TITLE
oci: use alternative registry

### PR DESCRIPTION
**Pull request by tigarmo**
Docker Hub is fast but the Ubuntu images are updated infrequently and
the registry itself has pull limits for anonymous users. This limit is
annoying because currently rockcraft does not caching of base images.
    
The caching is coming, but in the meantime switch to Amazon ECR. The
registry has newer images and is only a little bit slower (from testing)
than Docker Hub).

### CODE REVIEW

File: rockcraft/oci.py [hunk 1]
```diff
@@ -36,6 +36,12 @@
 
 logger = logging.getLogger(__name__)
 
+# Public Amazon Elastic Container Registry has images that are updated more frequently
+# and with less severe pull-rate limits than Docker Hub.
+ECR_URL = "public.ecr.aws/ubuntu"
```
Comment (SamirPS) [inline rockcraft/oci.py:41]: Why did you add this variable and not just do REGISTRY_URL = "public.ecr.aws/ubuntu" I don't see the value of this variable.

**Review by cjdcordeiro — APPROVED** [review]
lgtm, thanks

**Review by cmatsuoka — APPROVED** [review]
Looks good.
